sdk/core/Azure.Core/src/Messaging/CloudEventConverter.cs

```diff
@@ -0,0 +1,327 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Core;
+
+namespace Azure.Messaging
+{
+    /// <summary>
+    /// A custom converter that attributes the <see cref="CloudEvent"/> type.
+    /// This allows System.Text.Json to serialize and deserialize CloudEvents by default.
+    /// </summary>
+    internal class CloudEventConverter : JsonConverter<CloudEvent>
+    {
+        /// <summary>
+        /// Gets or sets the serializer to use for the data portion of the <see cref="CloudEvent"/>. If not specified,
+        /// JsonObjectSerializer is used.
+        /// </summary>
+        /// <inheritdoc cref="JsonConverter{CloudEvent}.Read(ref Utf8JsonReader, Type, JsonSerializerOptions)"/>
+        public override CloudEvent Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            JsonDocument requestDocument = JsonDocument.ParseValue(ref reader);
+            return DeserializeCloudEvent(requestDocument.RootElement, strict: true);
+        }
+
+        internal static CloudEvent DeserializeCloudEvent(JsonElement element, bool strict)
+        {
+            var cloudEvent = new CloudEvent();
+            foreach (JsonProperty property in element.EnumerateObject())
+            {
+                if (property.NameEquals(CloudEventConstants.Id))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        continue;
+                    }
+                    cloudEvent.Id = property.Value.GetString()!;
+                }
+                else if (property.NameEquals(CloudEventConstants.Source))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        continue;
+                    }
+                    cloudEvent.Source = property.Value.GetString()!;
+                }
+                else if (property.NameEquals(CloudEventConstants.Data))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
```

Comment:
I don't think you need to skip nulls here.

Reply:
Why do that? we would then end up with a BinaryData instance wrapping the JSON value of null as opposed to just a null BinaryData instance.

Reply:
`new CloudEvent(..., new BinaryData("null"), Format.Json);`